frontend/packages/pipelines-plugin/src/components/shared/workspaces/VolumeClaimTemplateLink.tsx

```diff
@@ -0,0 +1,41 @@
+import * as React from 'react';
+import { useK8sWatchResource } from '@console/internal/components/utils/k8s-watch-hook';
+import { K8sResourceKind } from '@console/internal/module/k8s';
+import { PersistentVolumeClaimModel } from '@console/internal/models';
+import { ResourceLink } from '@console/internal/components/utils';
+
+export interface VolumeClaimTemplateLinkProps {
+  labels: { [key: string]: string };
+  workspaceName: string;
+  namespace: string;
+}
+
+const VolumeClaimTemplateLink: React.FC<VolumeClaimTemplateLinkProps> = ({
+  labels,
+  workspaceName,
+  namespace,
+}) => {
+  const [pvcResource, loaded, loadError] = useK8sWatchResource<K8sResourceKind>({
+    kind: PersistentVolumeClaimModel.kind,
+    isList: false,
+    selector: { matchLabels: labels },
+  });
```

Comment:
I think you also need to look at the owner field 🤔 

Otherwise you're going to fetch the first PVC that matches a label and it's a coin flip if you get the one you want.

Suggested new path:
- Fetch all PVCs that match the label
- Look at the metadata.ownerReferences and trim down to the PLR / TR that matches
- List only the remaining PVCs

Reply:
Updated with this logic and also added tests